app/components/progress-updater.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+class ProgressUpdater {
+
+  constructor() {
+
+    this.state = {
+      playing: false,
+      progressObject: null,
+      progressPosition: '0%',
+      subjectViewer: null
+    };
+
+    this.progressUpdateListener = this.handleProgressUpdate.bind(this);
+    this.progressMarkerRenderer = this.renderProgressMarker.bind(this);
+    this.progressMonitoredObject = this.registerProgressingObject.bind(this);
+  }
+
+  setSubjectViewer(subjectViewer){
+    this.state.subjectViewer = subjectViewer;
+  }
+
+  updateSubjectViewer(){
+    if(this.state.subjectViewer){
+      this.state.subjectViewer.forceUpdate();
+    }
+  }
+
+  handleProgressUpdate(event) {
+    var xPosition = 100 * (this.state.progressObject.currentTime / this.state.progressObject.duration)
```

Comment:
I'm very uneasy about directly accessing the state of an external component here. The internal implementation of the progress object should be opaque, so that changes in implementation don't break components, like this one, that use that component. 

It would be better if the audio player passed down changes, as props, to a child component that rendered the spectrogram.